hammers/scripts/gpu_lease_stacking.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+'''
+.. code-block:: bash
+
+    gpu-lease-stacking {info, delete}
+
+Reclaims GPU nodes from leases that violate terms of use.
+
+* ``info`` to just display leases or actuall delete them with ``delete``
+'''
+import sys
+import argparse
+from pprint import pprint
+from datetime import datetime
+
+from hammers import MySqlArgs, osapi, query
+from hammers.slack import Slackbot
+from hammers.osrest.blazar import lease_delete
+
+
+def find_stacked_leases(leases):
+    """Return list of only the leases stacked on each other."""
+    stacked = []
+
+    for i in range(len(leases)):
+
+        start_date = leases[i][1]
+        end_date = leases[i][2]
+
+        if i > 0:
+            last_end_date = leases[i - 1][2]
+        else:
+            last_end_date = datetime.min
+
+        if i < len(leases) - 1:
+            next_start_date = leases[i + 1][1]
+        else:
+            next_start_date = datetime.max
+
+        stacked_previous = (start_date - last_end_date).days < 1
+        stacked_next = (next_start_date - end_date).days < 1
+
+        if stacked_previous or stacked_next:
+            stacked.append(leases[i])
```

Comment:
This means that one lease can be in there a bunch of times, which is OK because you do `set` later on, just calling it out. I guess it's tricky to make sure the first and last leases are included in the list.

Reply:
I used `set` because the sql query was outputting some duplicates which I didn't get to the bottom of. This is just checking if a given lease is part of a stack. I don't think this will produce duplicates because it's only appending the lease the loop is iterating over.